scripts/clean_osm_data.py

```diff
@@ -279,279 +293,382 @@ def finalize_lines_type(df_lines):
-    # if not int make int
-    if df_all_lines["cables"].dtype != int:
-        dropped_cables = [
-            x for x in dropped_cables_tags if x in df_all_lines["cables"].values
-        ]
-        if len(dropped_cables) != 0:
-            logger.info(
-                f"The lines with a cables tag in {set(dropped_cables)} will be dropped."
-            )
-
-        # map known non-numerical issues into a reasonable n_cables value
-        df_all_lines["cables"] = (
-            df_all_lines["cables"]
-            .map(cables_tag_to_n_cables)
-            .fillna(df_all_lines["cables"])
-        )
-        # HERE. "0" if cables "None", "nan" or "1"
-        df_all_lines.loc[
-            (df_all_lines["cables"] < "3") | df_all_lines["cables"].isna(), "cables"
-        ] = "0"
-
-        # there may be some non-known numerical issues
-        not_resolved_cables = (
-            ~pd.to_numeric(df_all_lines["cables"], errors="coerce")
-            .astype(float)
-            .apply(float.is_integer)
-        )
-        unknown_cables_tags = set(
-            df_all_lines.loc[not_resolved_cables]["cables"].values
-        )
-        if any(not_resolved_cables):
-            df_all_lines.drop(df_all_lines[not_resolved_cables].index, inplace=True)
-            logger.warning(
-                f"The lines with an unexpected cables tag value in {unknown_cables_tags} will be dropped."
-            )
-
-        df_all_lines["cables"] = df_all_lines["cables"].astype("int")
-
-    # downgrade 4 and 5 cables to 3...
-    if 4 or 5 in df_all_lines["cables"].values:
-        # Reason: 4 cables have 1 lighting protection cables, 5 cables has 2 LP cables - not transferring energy;
-        # see https://hackaday.com/2019/06/11/a-field-guide-to-transmission-lines/
-        # where circuits are "0" make "1"
-        df_all_lines.loc[
-            (df_all_lines["cables"] == 4) | (df_all_lines["cables"] == 5), "cables"
-        ] = 3
-
-    # one circuit contains 3 cable under a preliminary assumption of an AC line
-    df_all_lines.loc[df_all_lines["circuits"].isna(), "circuits"] = (
-        df_all_lines.loc[df_all_lines["circuits"].isna(), "cables"] / 3
+    return df
+
+
+def split_and_match_voltage_frequency_size(df):
+    """
+    Function to match the length of the columns in subset
+    by duplicating the last value in the column
+
+    The function does as follows:
+    1. First, it splits voltage and frequency columns by semicolon
+       For example, the following lines
+       row 1: '50', '220000
+       row 2: '50;50;50', '220000;380000'
+
+       become:
+       row 1: ['50'], ['220000']
+       row 2: ['50','50','50'], ['220000','380000']
+
+    2. Then, it harmonize each row to match the length of the lists
+       by filling the missing values with the last elements of each list.
+       In agreement to the example of before, after the cleaning:
+
+       row 1: ['50'], ['220000']
+       row 2: ['50','50','50'], ['220000','380000','380000']
```

Comment:
An alternative cleaning fo this case may be:
row 2: ['50','50'], ['220000','380000']

Using the following: when the length in frequency is higher than in voltage, we stay conservative and we drop a frequency value.
However, this is not frequent